scripts/check_compliance.py

```diff
@@ -668,6 +668,51 @@ def run(self):
             self.add_failure(msg.decode("utf-8"))
 
 
+class PyLint(ComplianceTest):
+    """
+    Runs pylint on all .py files, with a limited set of checks enabled. The
+    configuration is in the pylintrc file.
+    """
+    _name = "pylint"
+    _doc = "https://www.pylint.org/"
+
+    def run(self):
+        self.prepare(ZEPHYR_BASE)
+
+        # Path to pylint configuration file
+        pylintrc = os.path.join(os.path.dirname(__file__), "pylintrc")
+
+        # List of .py files added/modified by the commit(s).
+        #
+        # Skip create_board_img.py to work around a crash in pylint 2.2.2:
+        # https://github.com/PyCQA/pylint/issues/2906
+        py_files = git(
+            "diff", "--name-only", self.commit_range, "--", "*.py",
```

Comment:
Does this not fail to run when renaming or deleting a file? See ``--diff-filter` in other checks.

Reply:
Oops... fixed. Went with `--diff-filter=d` (everything but deleted). Looked like the rest would be safe to check, and maybe it could find stuff we haven't thought of.

Reply:
(Just for the record)

For renames that have been detected (depending on the curent similarity threshold), it seems `--name-only` prints only the new name. Interesting!